turbinia/workers/analysis/jupyter.py

```diff
@@ -0,0 +1,116 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for analysing Jupyter."""
+
+from __future__ import unicode_literals
+
+import os
+import re
+
+from turbinia import TurbiniaException
+from turbinia.evidence import ReportText
+from turbinia.lib import text_formatter as fmt
+from turbinia.workers import TurbiniaTask
+from turbinia.workers import Priority
+from turbinia.lib.utils import extract_files
+from turbinia.lib.utils import bruteforce_password_hashes
+
+
+class JupyterAnalysisTask(TurbiniaTask):
+  """Task to analyze a Jupyter Notebook config."""
+
+  def run(self, evidence, result):
+    """Run the Jupyter worker.
+
+    Args:
+        evidence (Evidence object):  The evidence to process
+        result (TurbiniaTaskResult): The object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+
+    # Where to store the resulting output file.
+    output_file_name = 'jupyter_analysis.txt'
+    output_file_path = os.path.join(self.output_dir, output_file_name)
+
+    # What type of evidence we should output.
+    output_evidence = ReportText(source_path=output_file_path)
+
+    # Read the config file.
+    # with open(evidence.local_path, 'r') as input_file:
+    jupyter_config = open(evidence.local_path, 'r').read()
+
+    # Extract the config and return the report
+    (report, priority, summary) = self.analyse_config(jupyter_config)
+    output_evidence.text_data = report
+    result.report_priority = priority
+    result.report_data = report
+
+    # Write the report to the output file.
+    with open(output_file_path, 'w') as fh:
+      fh.write(output_evidence.text_data.encode('utf8'))
+      fh.write('\n'.encode('utf8'))
+
+    # Add the resulting evidence to the result object.
+    result.add_evidence(output_evidence, evidence.config)
+    result.close(self, success=True, status=summary)
+
+    return result
+
+  def analyse_config(self, jupyter_config):
+    """Extract version from Jupyter configuration files.
+
+    Args:
+      config (str): configuration file content.
+
+    Returns:
+      str: The version of Jupyter.
+    """
+    import logging
+    findings = []
+    for line in jupyter_config.split('\n'):
+
+      if all(x in line for x in ['disable_check_xsrf', 'True']):
+        findings.append(fmt.bullet('XSRF protection is disabled.'))
+        continue
+      if all(x in line for x in ['allow_root', 'True']):
+        findings.append(fmt.bullet('Jupyter Notebook runs as admin.'))
+        continue
+      if 'NotebookApp.port' in line:
+        port = line.split('=')[1].replace(' ', '')
```

Comment:
This parsing is a bit risky since it assumes there will always be an equals with something following it on every line that contains 'NotebookApp.port'.    If that value is referenced in a comment or something without the second part of the line, this code would error out.  Maybe either use regexes or validate the output from `split` before using it?

Reply:
Temporarily removing the port check.